auth/auth.go

```diff
@@ -0,0 +1,7 @@
+package auth
+
+import (
+	"context"
+)
+
+type AuthFunc func(ctx context.Context) (context.Context, error)
```

Comment:
separate authentication from authorization 👍

> note: those two can be in the same package though 😄

```suggestion
type AuthenticationFunc func(ctx context.Context) (context.Context, error)
```

or

```suggestion
type AuthorizationFunc func(ctx context.Context) (context.Context, error)
```

depending on what you are doing 👍